### PR TITLE
Cleaner logging tests

### PR DESCRIPTION
**Pull request by mxgrey**
This PR takes advantage of `rcutils_logging_set_output_handler` to override the way log outputs are published within the logging test function. We prevent the messages from being printed and instead collect them in way that the test can access them and validate what is being produced.

This has two big advantages:
* We no longer get false "Error" and "Warning" messages printing out to the terminal when running these tests
* We can explicitly validate that the logger is behaving the way we expect

This branch builds directly off of #7, so that should be merged before this is reviewed.

### CODE REVIEW

**Comment by mxgrey** [comment]
I just realized a major issue with some of the changes in this PR.

When you specify a custom log output handler function to rcutils, your handler function will be given the format string (`"%s"`) along with the variadic arguments that the user passed in to fill into the format string. 

The variadic arguments are pretty much indecipherable from the Rust side, so in order to make sense of the log messages on the rclrs side, I skipped passing `("%s", msg)` to rcutils and instead simplified it to just passing in `msg` with no variadic arguments. 

This is fine in our tests since we can choose to just look at the message and ignore the variadic arguments. However this will be a problem when we're not testing because it's possible for someone to pass a string with a `%` character into their log message, and when rcl processes that according to its normal formatting rules, it can result in a segmentation fault or other undefined behavior. 

I'll need to tweak the PR to have slightly different behavior when compiled for tests vs compiled for real. A better solution would be to find a way to format a string from Rust given C variadic arguments, but I expect that to be an unreasonably difficult thing to do.

**Comment by mxgrey** [comment]
The safety issue is fixed by [89809d1](https://github.com/geoff-imdex/ros2_rust/pull/11/commits/89809d15980140c66ff6910db542a615dcb74e9f), so this PR should be good to go.

File: rclrs/src/logging/logging_configuration.rs [hunk 1]
```diff
@@ -55,3 +49,201 @@ impl Drop for LoggingLifecycle {
...
+            timestamp,
+            message,
+            logging_output,
+        );
+    }
+
+    /// Reset the logging output handler to the default one
+    pub(crate) fn reset_logging_output_handler() {
+        let _lifecycle = ENTITY_LIFECYCLE_MUTEX.lock().unwrap();
+        unsafe {
```
Comment (geoff-imdex) [inline rclrs/src/logging/logging_configuration.rs:242]: Sorry, very pedantic, but would it be possible to add the "safety" comment?

**Review by geoff-imdex — COMMENTED** [review]
As always, lots of great stuff (definitely helping my Rust learnings!).  One extremely pedantic comment for adding in the standard `safety` stuff.

File: rclrs/src/logging/logging_configuration.rs (hunk 1 above)
Comment (mxgrey) [reply rclrs/src/logging/logging_configuration.rs:242]: That's definitely not pedantic or something to apologize for, thanks for catching it. I wish clippy would flag missing safety comments.. it becomes easy to forget when dealing with language bindings that need dozens of `unsafe` blocks.

Fixed: [8c294d1](https://github.com/geoff-imdex/ros2_rust/pull/11/commits/8c294d1bce7efa35c779232d278c344a023fe52e)

**Comment by mxgrey** [comment]
Based on some discussions in the working group, I've updated `interval` to be called `throttle` instead, aligning the terminology with the other ROS client libraries.

**Comment by mxgrey** [comment]
I've also merged in the latest `main` branch from upstream. I think this PR should be good to go. Hopefully this will be the last one needed for the maintainer review.

**Pull request merged**